libbeat/outputs/outil/select.go

```diff
@@ -274,49 +303,27 @@ func buildSingle(cfg *common.Config, key string) (SelectorExpr, error) {
 	}
 
 	// 5. build selector from available fields
-	var sel SelectorExpr
+	var sel selectorExpr
 	if len(mapping.Table) > 0 {
-		if evtfmt.IsConst() {
-			str, err := evtfmt.Run(nil)
-			if err != nil {
-				return nil, err
-			}
-
-			str = mapping.Table[str]
-			if str == "" {
-				str = otherwise
-			}
-
-			if str == "" {
```

Comment:
I might be too early to click on approve. Don't we need to set `sel` to `nilSelector` if `otherwise` is an empty string?

Reply:
We have had this pattern in multiple locations. I moved some of the logic into the expression constructors. e.g. `constSelectorExpr` will check if `str` is "" and create a `nilSelector` or `constSelector`. This change makes code (hopefully) a little easier to read, while reducing duplication and chance of errors.